From 4c3856a09e30d0dfe9bc0cb2594e0d83daf82706 Mon Sep 17 00:00:00 2001
From: Charles Plessy <plessy@debian.org>
Date: Wed, 16 Oct 2013 08:13:01 +0900
Subject: [PATCH] Policy: on upgrades, recommend removing obsolete unchanged
 conf. files.

Wording: Paul Wise <pabs@debian.org>
Seconded: Jonathan Nieder <jrnieder@gmail.com>
Seconded: Charles Plessy <plessy@debian.org>
Closes: #707077
---
 debian/changelog         | 5 +++++
 policy.sgml              | 6 ++++--
 upgrading-checklist.sgml | 6 ++++++
 3 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/debian/changelog b/debian/changelog
index 103f90a..0ef998e 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -49,6 +49,11 @@ debian-policy (3.9.5.0) UNRELEASED; urgency=low
     Seconded: Steve Langasek <vorlon@debian.org>
     Seconded: Julien Cristau <jcristau@debian.org>
     Closes: #707183
+  * Policy: on upgrades, recommend removing obsolete unchanged conf. files.
+    Wording: Paul Wise <pabs@debian.org>
+    Seconded: Jonathan Nieder <jrnieder@gmail.com>
+    Seconded: Charles Plessy <plessy@debian.org>
+    Closes: #707077
   * debconf_spec: Document the 'escape' capability.
     Wording: Jonathan Nieder <jrnieder@gmail.com>
     Seconded: Charles Plessy <plessy@debian.org>
diff --git a/policy.sgml b/policy.sgml
index 113429b..138d320 100644
--- a/policy.sgml
+++ b/policy.sgml
@@ -9009,8 +9009,10 @@ ln -fs ../sbin/sendmail debian/tmp/usr/bin/runq
 		  package is purged.
 	      </item>
 	    </list>
-	    Obsolete configuration files without local changes may be
-	    removed by the package during upgrade.
+	    Obsolete configuration files without local changes should be
+	    removed by the package during upgrade.<footnote>
+	      The <prgn>dpkg-maintscript-helper</prgn> tool, available from the
+	      <package>dpkg</package> package, can help for this task.</footnote>
 	  </p>
 
 	  <p>
diff --git a/upgrading-checklist.sgml b/upgrading-checklist.sgml
index a77413d..f06ca80 100644
--- a/upgrading-checklist.sgml
+++ b/upgrading-checklist.sgml
@@ -62,6 +62,12 @@ Unreleased.
 <tag>9.1.1.8</tag>
   <item>The exception to the FHS for the <file>/selinux</file> was removed.
   </item>
+<tag>10.7.3</tag>
+  <item>During package upgrades, the removal of obsolete configuration files
+  without local changes is now recommended (<em>should</em>) instead of
+  suggested (<em>may</em>).  Note the <prgn>dpkg-maintscript-helper</prgn>
+  tool, available from the <package>dpkg</package> package since
+  <em>Wheezy</em>.
 <tag>10.10</tag>
   <item>The name of the files and directories installed by binary packages
   must be encoded in UTF-8 and should be restricted to ASCII when possible.
-- 
1.8.4.rc3

